File: test/csharp/EquivalenceComparerTests.cs
using Xunit;

namespace Jds.JsonEquivalence.Tests.CSharp
{
  /// <summary>
  ///   Tests validating the <see cref="EquivalenceComparer" /> API, as used by C#.
  /// </summary>
  public static class EquivalenceComparerTests
  {
    private const string MinifiedExample = @"{""name"":""Test""}";

    private const string PrettyPrintedExample = @"{
  ""name"" : ""Test""
}";

    private const string PrettyPrintedDifferingCasingExample = @"{
  ""name"" : ""TEST""
}";

    /// <summary>
    ///   Tests validating the <see cref="EquivalenceComparer.AreEquivalent" /> API, as used by C#.
    /// </summary>
    public class AreEquivalent
    {
      private static bool Act(string expected, string actual)
      {
        var options = new EquivalenceOptions(new EquivalenceValueOptions(false));
        return EquivalenceComparer.AreEquivalent(options, expected, actual);
      }

      [Theory]
      [InlineData(MinifiedExample, PrettyPrintedExample, true)]
      [InlineData(MinifiedExample, PrettyPrintedDifferingCasingExample, true)]
      public void IdentifiesEquivalence(string expected, string actual, bool shouldMatch)
      {
        Assert.Equal(shouldMatch, Act(expected, actual));
      }
    }

    /// <summary>
    ///   Tests validating the <see cref="EquivalenceComparer.AreStrictlyEquivalent" /> API, as used by C#.
    /// </summary>
    public class AreStrictlyEquivalent
    {
      private static bool Act(string expected, string actual)
      {
        return EquivalenceComparer.AreStrictlyEquivalent(expected, actual);
      }

      [Theory]
      [InlineData(MinifiedExample, PrettyPrintedExample, true)]
      [InlineData(MinifiedExample, PrettyPrintedDifferingCasingExample, false)]
      public void IdentifiesEquivalence(string expected, string actual, bool shouldMatch)
      {
        Assert.Equal(shouldMatch, Act(expected, actual));
      }
    }
  }
}

File: test/unit/EquivalenceComparerTests.fs
namespace Jds.JsonEquivalence.Tests.Unit

open Jds.JsonEquivalence
open Xunit

module EquivalenceComparerTests =
    module AreEquivalent =
        [<Fact>]
        let ``Identifies equivalence``() =
            let options = EquivalenceOptions(EquivalenceValueOptions(false))
            let actual =
                EquivalenceComparer.AreEquivalent
                    (options, TestData.NestedObjectCases.sourceJson, TestData.NestedObjectCases.childPropertyReordered)
            Assert.True(actual)

    module AreStrictlyEquivalent =
        [<Fact>]
        let ``Identifies equivalence``() =
            let actual =
                EquivalenceComparer.AreStrictlyEquivalent
                    (TestData.NestedObjectCases.sourceJson, TestData.NestedObjectCases.deepStringPropertyDiffers)
            Assert.False(actual)
